cmake/compilerFlags.cmake

```diff
@@ -169,6 +169,7 @@ if(MSVC)
 
     # Object Level Parallelism
     add_compile_options(/MP)
+    add_compile_options(/utf-8)

```

Comment:
Do we need something equivalent for MinGW gcc/clang, or nothing is required there? I.e. I don't see `-DUNICODE -D_UNICODE` defined anywhere that would give us string-wstring etc. on MinGW, if presumably `/utf-8` does this for MSVC? If these are equivalent, maybe we prefer the compiler independent definitions?

Also maybe add a separate comment, the one above does not pertain to this line?

Reply:
The MSYS2/MinGW CI jobs indeed failed. I have to check locally with that environment to check what to do there. The new test added is at the moment failing on that platform:
https://github.com/Exiv2/exiv2/runs/5154900058?check_suite_focus=true

Reply:
I personally would like to have a portable codebase that runs on Windows without assuming the compiler. There are many projects out there that rely on MinGW provided binaries.

Reply:
Me too. I would like to take a pragmatic approach and fix one thing at each time. We can focus on this PR on fixing #1996 for MSVC and later investigate how to do it with MinGW.

I think that by following that approach the build with MinGW would remain the same (we would not be breaking anything).

However, I need to check how I could skip the new test on Win+MinGW builds

Reply:
I don't think this is the way to go then if this is MSVC specific.

I still don't get why this approach won't (or currently doesn't) work:

- have main() w/ char* argc (i.e. assume multi-byte UTF-8 input for all arguments on all platforms)
- parse arguments as multi-byte UTF-8 char* only on all platforms (i.e. as today)
- on Windows, convert **only filenames** w/ existing s2ws() when needed to open the I/O w/ Windows wchar APIs

The only sticking point, as you say, might be what the different consoles encode, I haven't checked that thoroughly.